enzyme/benchmarks/ReverseMode/ba/Makefile.make

```diff
@@ -3,21 +3,23 @@
 .PHONY: clean
 
 dir := $(abspath $(lastword $(MAKEFILE_LIST))/../../../..)
+CXX = clang++
+OPT = opt
 
 clean:
 	rm -f *.ll *.o results.txt results.json
 
 %-unopt.ll: %.cpp
-	clang++ $(BENCH) $(PTR) $^ -pthread -O2 -fno-vectorize -fno-slp-vectorize -ffast-math -fno-unroll-loops -o $@ -S -emit-llvm
+	$(CXX) $(BENCH) $(PTR) $^ -pthread -O2 -fno-vectorize -fno-slp-vectorize -ffast-math -fno-unroll-loops -o $@ -S -emit-llvm
 
 %-raw.ll: %-unopt.ll
-	opt $^ $(LOAD) $(ENZYME) -o $@ -S
+	$(OPT) $^ $(LOAD) $(ENZYME) -o $@ -S
 
 %-opt.ll: %-raw.ll
-	opt $^ -o $@ -S
+	$(OPT) $^ -o $@ -S
 
 ba.o: ba-opt.ll
-	clang++ $(BENCH) -pthread -O2 $^ -I /usr/include/c++/11 -I/usr/include/x86_64-linux-gnu/c++/11 -O2 -o $@ $(BENCHLINK) -lpthread -lm -L /usr/lib/gcc/x86_64-linux-gnu/11
+	$(CXX) $(BENCH) -pthread -O2 $^ -I /usr/include/c++/11 -I/usr/include/x86_64-linux-gnu/c++/11 -O2 -o $@ $(BENCHLINK) -lpthread -lm -L /usr/lib/gcc/x86_64-linux-gnu/11
```

Comment:
Is this change welcome here? I'm running like `CXX=/home/jeka2967/src/rust/build/host/llvm/bin/clang++ OPT=/home/jeka2967/src/rust/build/host/llvm/bin/opt ninja bench-gmm-reverse` to use the same toolchain for Rust and C++. In principle this information could be based on `LLVM_DIR` in the CMake, but I don't know how to access that from this makefile and I think this flexibility is desirable.

Reply:
CXX could cause issues with unhelpful error messages if people accidentally have GCC behind CXX, plus your call command is extremely verbose. I have 5 different  (incompatible) LLVM versions so I don't want to use export and right now running `ninja bench-gmm-reverse` seems to always pick the right one, so I'm hesitant to regress the usability so much. So if anything I would just change it to use the LLVM_DIR from cmake to automatically use the right one more confidently and leave it unchanged till then. I'll update the other benchmarks to include the env variable and update the Makefile.makeafter, unless you beat me to it.

Reply:
If it's working for you now, then nothing will change with this patch. Perhaps the benchmarking system is setting `PATH` so that `clang++` is pulled from `$LLVM_DIR/bin`? I saw a failure due to what looked like mismatch between LLVM used to build Enzyme and the default `clang++` so I put this in and it passed. I didn't dig to see if that was really the cause.

Reply:
so the path should be automatically set to point to the directory containing Enzyme's LLVM binaries. So it should always point to the right things, assuming there is something named clang there.

If we were to do this change, probably we should set it to the autopopulated path to llvm clang directly.
For example https://github.com/EnzymeAD/Enzyme/blob/fffdd2c49cd342d2ba1d4dff08bcd21db1390e58/enzyme/benchmarks/lit.site.cfg.py.in#L49 or https://github.com/EnzymeAD/Enzyme/blob/fffdd2c49cd342d2ba1d4dff08bcd21db1390e58/enzyme/test/lit.site.cfg.py.in#L71

Reply:
here is where the path is set: https://github.com/EnzymeAD/Enzyme/blob/fffdd2c49cd342d2ba1d4dff08bcd21db1390e58/enzyme/benchmarks/lit.cfg.py#L37

Reply:
Okay, thanks for pointing out where this is handled. I'm not reproducing the problem I saw earlier and `readelf -p .comment ba.o` reports the Rust toolchain (based on my `LLVM_DIR`) so it seems to be working. I'll drop this commit since it seems unnecessary.